compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp

```diff
@@ -484,6 +484,9 @@ void addGPUTileAndFusePassPipeline(OpPassManager &funcPassManager,
     options.redundantHoisting = false;
     funcPassManager.addPass(createOptimizeVectorTransferPass());
   }
+  // Step 10. Hoist statically bounded allocations. Need to pad dynamic
+  // allocations at this stage to make sure they can be hoisted.
+  funcPassManager.addPass(createPadDynamicAllocPass());
```

Comment:
THis pass is already run on LowerToLLVMGPU passes. So this + hoist statically bound allocations could just run there.